src/pybind/mgr/mgr_module.py

```diff
@@ -733,10 +736,31 @@ def get_all_perf_counters(self, prio_limit=PRIO_USEFUL):
                     if counter_schema['priority'] < prio_limit:
                         continue
 
-                    counter_info = counter_schema
-                    counter_info['value'] = self.get_latest(service['type'], service['id'],
-                                                            counter_path)
-                    result[svc_full_name][counter_path] = counter_info
+                    # Deal with long running avgs properly
+                    if counter_schema['type'] & self.PERFCOUNTER_LONGRUNAVG:
+                        counter_sum = dict(counter_schema)
+                        counter_sum['value' ] = self.get_latest(
+                            service['type'],
+                            service['id'],
+                            counter_path
+                        )
+                        result[svc_full_name][counter_path + '_sum'] = counter_sum
+
+                        counter_avgcount = dict(counter_schema)
+                        # The avgcount is always a regular integer, not time
+                        counter_avgcount['type'] = self.PERFCOUNTER_U64 + self.PERFCOUNTER_LONGRUNAVG
+                        counter_avgcount['value' ] = self.get_latest(
+                            service['type'],
+                            service['id'],
+                            counter_path,
+                            avgcount = True
+                        )
+                        result[svc_full_name][counter_path + '_count'] = counter_avgcount
```

Comment:
The values in `result` are already dicts, so wouldn't it be simpler to have 'sum' and 'count' members in the same dict, rather than having separate _count and a _sum entries?